### PR TITLE
Code checks

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Get storage client generation working as part of CI:
- Use Node 10.15.0 instead of 12.x.
- Update Generate script to install and use global autorest@beta.
- Prevent git EOL warnings which were triggering exceptions in CodeChecks

### CODE REVIEW

File: sdk/storage/generate.ps1 [hunk 1]
```diff
@@ -1,16 +1,17 @@
 pushd $PSScriptRoot/Azure.Storage.Common/swagger/Generator/
 npm install
+npm install -g autorest@beta
```
Comment (pakrym) [inline sdk/storage/generate.ps1]: So local installations of autorest do not work? Did we file a bug about it?

**Review by pakrym — APPROVED** [review]

File: eng/CodeChecks.ps1 [hunk 2]
```diff
@@ -72,18 +72,19 @@ try {
...
     Write-Host "Re-generating clients"
     Invoke-Block {
-        # https://github.com/Azure/azure-sdk-for-net/issues/8584
-        # & $repoRoot\storage\generate.ps1
+        & $repoRoot\storage\generate.ps1
     }
 
     Write-Host "git diff"
+    # prevent warning related to EOL differences which triggers an exception for some reason
+    & git config core.safecrlf false
```
Comment (weshaggard) [inline eng/CodeChecks.ps1]: I'm not sure I like this script automatically changing someone's local git config. If we want this option just for the git diff call pass the config directly with that command instead.

Comment (JoshLove-msft) [reply eng/CodeChecks.ps1]: I don't think there is a way to pass this as an argument to diff call. I can at least just lookup and then restore the value of core.safecrlf.

File: sdk/storage/generate.ps1 (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/storage/generate.ps1]: It seemed to still use the old core version when depending on the local dependency. I'm not sure that this is a bug, as you can specify the core version to use in the script. In any case, our Contributing.md recommends the global installation so it makes sense to be consistent here.

File: sdk/storage/generate.ps1 [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
 cd $PSScriptRoot/Azure.Storage.Blobs/swagger/
-npx autorest --use=$PSScriptRoot/Azure.Storage.Common/swagger/Generator/ --verbose
+autorest --use=$PSScriptRoot/Azure.Storage.Common/swagger/Generator/ --verbose
```
Comment (weshaggard) [inline sdk/storage/generate.ps1]: Shouldn't this be `autorest-beta`? @MiYanni

Comment (MiYanni) [reply sdk/storage/generate.ps1]: If I'm understanding which version they are using, they are using the one we made for PowerShell, which wasn't `autorest-beta` yet.

Comment (JoshLove-msft) [reply sdk/storage/generate.ps1]: Using `autorest-beta` doesn't work.

Comment (weshaggard) [reply sdk/storage/generate.ps1]: OK I get confused by "autorest@beta" vs "autorest-beta" :) I keep thinking they should be the same but they aren't.

**Review by weshaggard — APPROVED** [review]

File: sdk/storage/generate.ps1 (hunk 3 above)
Comment (MiYanni) [reply sdk/storage/generate.ps1]: Yeah. `autorest@beta` uses the `autorest` command while `@autorest/autorest` uses `autorest-beta`. I know... but this stuff is temporary.

**Review by seanmcc-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/swagger/Generator/package.json [hunk 4]
```diff
@@ -20,7 +20,6 @@
     "@types/mocha": "5.2.5",
     "@types/node": "10.12.19",
     "@types/semver": "5.5.0",
-    "autorest": "^3.0.5231",
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Common/swagger/Generator/package.json]: Why are we removing this?

Comment (pakrym) [reply sdk/storage/Azure.Storage.Common/swagger/Generator/package.json]: Using autorest via npx didn't work so there is no need in local installation.

**Pull request merged**